integration-tests/bats/config.bats

```diff
@@ -186,4 +186,25 @@ teardown() {
     run dolt sql -q "SET @@dolt_repo_$$_head = COMMIT('-a', '-m', 'updated stuff')"
     [ "$status" -eq 1 ]
     [[ "$output" =~ "Aborting commit due to empty committer email. Is your config set" ]] || false
+}
+
+@test "config: Set default init branch" {
+    dolt config --global --add user.name "bats tester"
+    dolt config --global --add user.email "joshn@doe.com"
+
+    dolt config --global --add init.default_branch "main"
+    run dolt config --list
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "init.default_branch = main" ]]
+
+    dolt init
```

Comment:
an argument to init that lets you specify the branch makes this more accessible for users.